### PR TITLE
Fixed and Updated Section Due Dates

### PR DESCRIPTION
**Pull request by alexinman**
Fixes all bugs mentioned in #1676 - Section Due Dates don't work

Section due dates now correctly work with the addition of being able to collect a single section's submissions at a time. 

Also moves the 'Collect All Submissions' link from the header to be a button above the submission table and is replaced by a 'Collect Section Submissions' button when there is at least one section.

This is what the submission page should look like with at least one section:
screen shot 2016-02-19 at 4 20 22 pm
And this is what it should look like without any sections:
screen shot 2016-02-19 at 4 41 58 pm

### CODE REVIEW

File: app/controllers/submissions_controller.rb [hunk 1]
```diff
@@ -218,23 +218,61 @@ def collect_and_begin_grading
...
+      end
+        
+      if collect_submissions_for_section(id, assignment, errors) > 0
+        successes.push(Section.find(id).name)
+      else
+        noSubmissions.push(Section.find(id).name)
+      end
+    end
+    if successes.length > 0
+      plural = successes.length > 1 ? 's' : ''
```
Comment (david-yz-liu) [inline app/controllers/submissions_controller.rb]: Remove the logic for pluralization and final "and". It's fine to just have section lists that look like

```
Sections collected: s1, s2, s3
```

File: app/views/submissions/_submissions_table.js.jsx.erb [hunk 2]
```diff
@@ -301,9 +307,76 @@
...
+                  <option value={"<%= ids.join(',') %>"}><%= name %></option>
+                <% end %>
+              </select>
+              <button onClick={this.collectSectionSubmissions}
+                      data-confirm='<%= t('collect_submissions.results_loss_warning') %>'>
+                <%= j raw I18n.t('collect_submissions.collect_section') %>
+              </button>
+            <% end
+          else %>
+            <button onClick={this.collectAllSubmissions}
```
Comment (david-yz-liu) [inline app/views/submissions/_submissions_table.js.jsx.erb]: This button should only appear for Admins, but not TAs.

File: app/views/submissions/_submissions_table.js.jsx.erb [hunk 3]
```diff
@@ -301,9 +307,76 @@
...
+        }.bind(this)
+      });
+    },
     render: function() {
       return (
         <div className='react-release-marks'>
+          <% if Section.all.count > 0 %>
+            <% sections = Hash.new
+            all_available_sections = Array.new
+            Section.all.each do |section|
```
Comment (david-yz-liu) [inline app/views/submissions/_submissions_table.js.jsx.erb]: Please put this logic into the controller.

File: app/controllers/submissions_controller.rb [hunk 4]
```diff
@@ -278,15 +316,38 @@ def browse
...
+          end
+          section_due_dates[collection_time].push(section.name)
+        end
+        section_due_dates.each do |collection_time, sections|
+          plural = sections.length > 1 ? 's' : '';
+          after = collection_time != now ? '_after' : ''
+          sections = sections.join(', ').reverse
+                             .sub(' ,', t('and').reverse)
+                             .reverse
+          flash_now(:notice, t('browse_submissions.grading_can_begin' +
```
Comment (david-yz-liu) [inline app/controllers/submissions_controller.rb]: Don't dynamically generate internationalization keys. Makes it harder to search for.

File: app/models/assignment.rb [hunk 5]
```diff
@@ -894,6 +898,19 @@ def add_graders_to_criterion(criterion_name, graders)
 
     criterion.add_tas_by_user_name_array(graders)
   end
+
+  # Returns the groupings of this assignment associated with the given section
+  def section_groupings(section)
+    section_groupings = Array.new
```
Comment (david-yz-liu) [inline app/models/assignment.rb]: Use select (http://ruby-doc.org/core-2.2.0/Array.html#method-i-select)

Comment (david-yz-liu) [inline app/models/assignment.rb]: @alexinman good work, except you can delete this line now

Comment (alexinman) [inline app/models/assignment.rb]: Yes, I noticed that after I committed/pushed. Removing that line is in my next commit. I just haven't pushed yet.

Comment (david-yz-liu) [inline app/models/assignment.rb]: I'd like to merge in this pull request ASAP, so can you please make that commit, and leave the test writing for a new pull request?

**Pull request merged**